fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/AddMinMax.java

```diff
@@ -326,7 +326,10 @@ private Map<Expression, MinMaxValue> getExprMinMaxValues(UnknownValue valueDesc)
         for (int i = 1; i < sourceValues.size(); i++) {
             // process in sourceValues[i]
             Map<Expression, MinMaxValue> minMaxValues = getExprMinMaxValues(sourceValues.get(i));
-            for (Map.Entry<Expression, MinMaxValue> entry : minMaxValues.entrySet()) {
+            List<Map.Entry<Expression, MinMaxValue>> minMaxValueList = minMaxValues.entrySet().stream()
+                    .sorted((a, b) -> Integer.compare(a.getValue().exprOrderIndex, b.getValue().exprOrderIndex))
+                    .collect(Collectors.toList());
+            for (Map.Entry<Expression, MinMaxValue> entry : minMaxValueList) {
```

Comment:
add comment to explain why need sort by expr id

Reply:
during mergging sourceValues[i] into result,  we want to keep the relative order of values of sourceValues[i] if they don't  in result.  for example if a's exprOrderIndex < b's exprOrderIndex,  then in result, a's new exprOrderIndex also < b's new exprOrderIndex.

also add this comment in code.

Reply:
add these comment into code please

Reply:
had add